baseplate/observers/metrics.py

```diff
@@ -101,7 +121,9 @@ def on_child_span_created(self, span: Span) -> None:
 class MetricsLocalSpanObserver(SpanObserver):
     def __init__(self, batch: metrics.Batch, span: Span, sample_rate: float = 1.0):
         self.batch = batch
-        self.timer = batch.timer(typing.cast(str, span.component_name) + "." + span.name)
+        self.timer = batch.timer(
```

Comment:
can probably f-string this now: `f"{span.component_name}.{span.name}"

if mypy still catches the potential `None` there the cast can go inside the f-string (:nauseated_face:)